packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/javamodel/JavaPackage.java

```diff
@@ -359,21 +360,33 @@ public final void addGraalVmConfig(String groupId, String artifactId, GraalVmCon
         }
     }
 
-    protected boolean checkDuplicateFile(String filePath) {
-        if (filePaths.contains(filePath)) {
-//            throw new IllegalStateException(String.format("Name conflict for output file '%1$s'.", filePath));
-            logger.warn(String.format("Name conflict for output file '%1$s'.", filePath));
-            return true;
-        }
-        return false;
-    }
-
     public void addJsonMergePatchHelper(List<ClientModel> models) {
         JavaFile javaFile
             = javaFileFactory.createSourceFile(settings.getPackage(settings.getImplementationSubpackage()),
                 ClientModelUtil.JSON_MERGE_PATCH_HELPER_CLASS_NAME);
+        this.checkDuplicateFile(javaFile.getFilePath());
         Templates.getJsonMergePatchHelperTemplate().write(models, javaFile);
         this.checkDuplicateFile(javaFile.getFilePath());
         javaFiles.add(javaFile);
     }
+
+    public void addTypeSpecMetadata(TypeSpecMetadata typeSpecMetadata) {
+        final String filePath = "src/main/resources/META-INF/" + typeSpecMetadata.getArtifactId() + "_metadata.json";
+        try {
+            TextFile textFile = new TextFile(filePath, typeSpecMetadata.toJsonString());
+            this.checkDuplicateFile(textFile.getFilePath());
+            textFiles.add(textFile);
+        } catch (IOException e) {
+            logger.warn("Failed to write metadata file {}", filePath);
+        }
+    }
+
+    protected boolean checkDuplicateFile(String filePath) {
+        if (filePaths.contains(filePath)) {
+//            throw new IllegalStateException(String.format("Name conflict for output file '%1$s'.", filePath));
```

Comment:
nit; maybe we want to delete this unused line?

Reply:
Thanks. I will change it to a comment (about why we decided not to fail on such condition).